test/tinytest.c

```diff
@@ -549,27 +550,25 @@ tinytest_main(int c, const char **v, struct testgroup_t *groups)
 		struct testgroup_t *group = &groups[i];
 		for (j = 0; group->cases[j].name; ++j) {
 			struct testcase_t *testcase = &group->cases[j];
-			int attempts = opt_retries;
+			int attempts = (testcase->flags & TT_RETRIABLE) ? opt_retries: 1;
```

Comment:
```suggestion
			int attempts = (testcase->flags & TT_RETRIABLE) ? opt_retries : 1;
```

Reply:
And now `--retries 0` works as infinite number of retries, not good.

Reply:
Ok, I've fixed it, but I cannot push to this topic branch, seems that you forbid this.